components/browser/session/src/test/java/mozilla/components/browser/session/SessionManagerTest.kt

```diff
@@ -972,4 +972,68 @@ class SessionManagerTest {
 
         manager.remove(session, selectParentIfExists = true)
     }
+
+    @Test
+    fun `SessionManager will select nearby private session if selected private session gets removed`() {
+        val manager = SessionManager(mock())
+        assertNull(manager.selectedSession)
+
+        val private1 = Session("https://example.org/private1", private = true)
+        manager.add(private1)
+
+        val regular1 = Session("https://www.mozilla.org", private = false)
+        manager.add(regular1)
+
+        val regular2 = Session("https://www.firefox.com", private = false)
+        manager.add(regular2)
+
+        val private2 = Session("https://example.org/private2", private = true)
+        manager.add(private2)
+
+        val private3 = Session("https://example.org/private3", private = true)
+        manager.add(private3)
+
+        manager.select(private2)
+        manager.remove(private2)
+        assertEquals(private3, manager.selectedSession)
+
+        manager.remove(private3)
+        assertEquals(private1, manager.selectedSession)
+
+        // Removing the last private session should cause a regular session to be selected
+        manager.remove(private1)
+        assertEquals(regular2, manager.selectedSession)
+    }
+
+    @Test
+    fun `SessionManager will select nearby non-private session if selected non-private session gets removed`() {
+        val manager = SessionManager(mock())
+        assertNull(manager.selectedSession)
+
+        val regular1 = Session("https://www.mozilla.org", private = false)
+        manager.add(regular1)
+
+        val private1 = Session("https://example.org/private1", private = true)
+        manager.add(private1)
+
+        val private2 = Session("https://example.org/private2", private = true)
+        manager.add(private2)
+
+        val regular2 = Session("https://www.firefox.com", private = false)
+        manager.add(regular2)
+
+        val regular3 = Session("https://www.firefox.org", private = false)
+        manager.add(regular3)
+
+        manager.select(regular2)
+        manager.remove(regular2)
+        assertEquals(regular3, manager.selectedSession)
+
+        manager.remove(regular3)
+        assertEquals(regular1, manager.selectedSession)
+
+        // Removing the last regular session should NOT cause a private session to be selected
+        manager.remove(regular1)
+        assertNull(manager.selectedSession)
```

Comment:
I just compared that with Fennec and saw that it indeed does not select a regular session in this case. However they create a new regular tab. So I wonder if we should invoke the `defaultSession` lambda in this case and select the new session?

Reply:
Good idea, yes! I will update this.

Reply:
@pocmo actually, this already works, because if we return NO_SELECTION from `recalculateSelectionIndex` we invoke the defaultsession lambda: https://github.com/mozilla-mobile/android-components/blob/master/components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt#L292

Reply:
Perfect :)

Reply:
Can we write a test for that? 🙃

Reply:
Already on it.

Reply:
@pocmo test added now!